plugin/evm/limitorders/limit_order_tx_processor.go

```diff
@@ -111,14 +112,15 @@ func (lotp *limitOrderTxProcessor) ExecuteMatchedOrdersTx(incomingOrder LimitOrd
 }
 
 func (lotp *limitOrderTxProcessor) executeLocalTx(contract common.Address, contractABI abi.ABI, method string, args ...interface{}) error {
-	nonce := lotp.txPool.GetOrderBookTxNonce(common.HexToAddress(userAddress1)) // admin address
+	address, _ := getAddressFromPrivateKey(lotp.validatorPrivateKey)
```

Comment:
can we cache the address to avoid having to derive everytime?

Reply:
made changes. Please review